himbaechel/himbaechel_api.h

```diff
@@ -109,10 +109,13 @@ struct HimbaechelAPI
     // Graphics
     virtual void drawBel(std::vector<GraphicElement> &g, GraphicElement::style_t style, IdString bel_type, Loc loc) {};
 
-    virtual void drawWire(std::vector<GraphicElement> &g, GraphicElement::style_t style, Loc loc, IdString wire_type, int32_t tilewire, IdString tile_type) {};
+    virtual void drawWire(std::vector<GraphicElement> &g, GraphicElement::style_t style, Loc loc, IdString wire_type,
+                          int32_t tilewire, IdString tile_type) {};
 
-    virtual void drawPip(std::vector<GraphicElement> &g,GraphicElement::style_t style, Loc loc,
-                WireId src, IdString src_type, int32_t src_id, WireId dst, IdString dst_type, int32_t dst_id) {};
+    virtual void drawPip(std::vector<GraphicElement> &g, GraphicElement::style_t style, Loc loc, WireId src,
+                         IdString src_type, int32_t src_id, WireId dst, IdString dst_type, int32_t dst_id) {};
+
+    virtual void drawGroup(std::vector<GraphicElement> &g, IdString group_type, Loc loc) {};
```

Comment:
@gatecat  Should I change interface by adding group_name or even better we can just propagating GroupId and let it handled depending of implementation ?
Was thinking about this API and for example in case of Ice40 there would be multiple switchboxes of same type, also we can represent warious muxes groups like this and those are drawn same way but on different places depending both on type but also on id.

Reply:
I think just GroupId is fine

Reply:
Thanks, updated code accordingly